c/meterpreter/source/extensions/priv/service.c

```diff
@@ -172,3 +172,47 @@ DWORD service_destroy( char * cpName )
 
 	return dwResult;
 }
+
+/*
+ * Query service state.
+ */
+DWORD query_service_status( char* cpName, DWORD* dwState )
+{
+	DWORD dwResult  = ERROR_SUCCESS;
+	HANDLE hManager = NULL;
+	HANDLE hService = NULL;
+	DWORD dwBytes;
+	SERVICE_STATUS_PROCESS procInfo;
+
+	do
+	{
+		if( !cpName )
+			BREAK_WITH_ERROR( "[SERVICE] query_service_status. cpName is NULL", ERROR_INVALID_HANDLE );
+
+		hManager = OpenSCManagerA( NULL, NULL, SC_MANAGER_ALL_ACCESS );
+		if( !hManager )
+			BREAK_ON_ERROR( "[SERVICE] query_service_status. OpenSCManagerA failed" );
+
+		hService = OpenServiceA( hManager, cpName, DELETE ); 
```

Comment:
Shouldn't we be calling DeleteService from https://learn.microsoft.com/en-us/windows/desktop/api/Winsvc/nf-winsvc-deleteservice here if we intend to delete a service like you seem to be doing here?

Reply:
It's updated. Are you sure? 
https://github.com/rapid7/metasploit-payloads/blob/29d6b431742f787096342a416c82dcb3054cc114/c/meterpreter/source/extensions/priv/service.c#L192-L198

Reply:
updated by this commit [here](https://github.com/rapid7/metasploit-payloads/pull/625/commits/f260c5f4c7cbac5487de08633fd0e2d452b772e3)